staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go

```diff
@@ -94,6 +94,10 @@ func (r *REST) Delete(ctx context.Context, name string, options *metav1.DeleteOp
 
 	// upon first request to delete, add our finalizer and then delegate
 	if crd.DeletionTimestamp.IsZero() {
+		if err := deleteValidation(obj); err != nil {
```

Comment:
if we're passing the object that is being deleted, we need to pass the one seen inside the GuaranteedUpdate callback... otherwise if there's a conflict, and the callback is called again with the actual latest object, admission won't see the actual object being deleted

Reply:
this is still outstanding